src/Connections/FeedConnection.ts

```diff
@@ -182,6 +185,15 @@ export class FeedConnection extends BaseConnection implements IConnection {
     }
 
     public async handleFeedEntry(entry: FeedEntry): Promise<void> {
+        // We will need to tidy this up.
+        if (this.state.template?.includes('$SUMMARY') && entry.summary) {
```

Comment:
How about using a full-word regex match on `$SUMMARY` to avoid potentially over-matching?
```suggestion
        if (this.state.template?.match(/\$SUMMARY\b/) && entry.summary) {
```